pkg/controller/direct/discoveryengine/mapper.go

```diff
@@ -65,7 +68,16 @@ func DiscoveryEngineEngineSpec_FromProto(mapCtx *direct.MapContext, in *pb.Engin
 	out.DisableAnalytics = direct.LazyPtr(in.GetDisableAnalytics())
 
 	for _, dataStoreID := range in.DataStoreIds {
-		out.DataStoreRefs = append(out.DataStoreRefs, &krm.DiscoveryEngineDataStoreRef{External: dataStoreID})
+		tokens := strings.Split(in.Name, "/")
```

Comment:
This (and the similar function below) is the trickiest thing in this PR, IMO.  The GCP API expects only the data_store_id, and implicitly resolves the project / location / collection because (I guess) we can't go across projects / locations / collections.